pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -1743,6 +1752,12 @@ private void updateCurrentDocumentCountMetrics() {
           rowsConsumed, consumedRate, _currentOffset, _numRowsConsumed, _numRowsIndexed);
       _lastConsumedCount = _numRowsConsumed;
       _lastLogTime = now;
+      if (_filteredMessageOffsets.size() > 0) {
+        if (_trackFilteredMessageOffsets) {
+          _segmentLogger.info("Filtered events with offsets: {}", _filteredMessageOffsets);
+        }
```

Comment:
self-review: adding a nested-if instead of combining it to ensure that _filteredMessageOffsets gets clear every time just in case in future someone changes the logic of pushing to _filteredMessageOffsets.